plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java

```diff
@@ -1007,13 +1008,26 @@ public boolean isS3SelectPushdownEnabled()
     }
 
     @Config("hive.s3select-pushdown.enabled")
-    @ConfigDescription("Enable query pushdown to AWS S3 Select service")
+    @ConfigDescription("Enable query pushdown to JSON files using the AWS S3 Select service")
     public HiveConfig setS3SelectPushdownEnabled(boolean s3SelectPushdownEnabled)
     {
         this.s3SelectPushdownEnabled = s3SelectPushdownEnabled;
         return this;
     }
 
+    public boolean isS3SelectExperimentalPushdownEnabled()
+    {
+        return s3SelectExperimentalPushdownEnabled;
+    }
+
+    @Config("hive.s3select-pushdown.experimental-pushdown-enabled")
+    @ConfigDescription("Enable query pushdown to TEXTFILE tables using the AWS S3 Select service. Requires 'hive.s3select-pushdown.enabled' also be set.")
```

Comment:
> Omit Requires 'hive.s3select-pushdown.enabled' also be set. here. It's kind of obvious.

Sure

> Also, it's possible to enforce using @AssertTrue annotation (add a test, since recent changes from javax to jakarta broke some validations)

I'd rather not require this, as the main toggle can also be enabled/disabled using a session property.